File: problems/18_bit_manipulation/0191_number_of_1_bits/solution.py
"""
Problem: 0191. Number of 1 Bits
"""


class Solution:
    def hammingWeight(self, n: int) -> int:
        c = 0
        while n:
            n &= n - 1
            c += 1
        return c


if __name__ == "__main__":
    sol = Solution()
    assert sol.hammingWeight(0b00000000000000000000000000001011) == 3
    assert sol.hammingWeight(0b00000000000000000000000010000000) == 1
    assert sol.hammingWeight(0b11111111111111111111111111111101) == 31
    assert sol.hammingWeight(0) == 0
    assert sol.hammingWeight(7) == 3
    print("All tests passed!")

File: problems/18_bit_manipulation/0191_number_of_1_bits/solution.dart
// 0191. Number of 1 Bits
int hammingWeight(int n) {
  var c = 0;
  while (n != 0) {
    n &= n - 1;
    c++;
  }
  return c;
}
void main() {
  if (hammingWeight(11) != 3) throw 'a';
  if (hammingWeight(128) != 1) throw 'b';
  if (hammingWeight(0) != 0) throw 'c';
  if (hammingWeight(7) != 3) throw 'd';
  if (hammingWeight(1023) != 10) throw 'e';
  print('All tests passed!');
}
